nomad/structs/config/consul.go

```diff
@@ -359,6 +401,74 @@ func (c *ConsulConfig) Copy() *ConsulConfig {
 	if nc.AllowUnauthenticated != nil {
 		nc.AllowUnauthenticated = pointer.Of(*nc.AllowUnauthenticated)
 	}
+	if nc.UseIdentity != nil {
+		nc.UseIdentity = pointer.Of(*nc.UseIdentity)
+	}
```

Comment:
We also need to copy `ServiceIdentity` and `TemplateIdentity`.

Reply:
corrected in [7a0f84f](https://github.com/hashicorp/nomad/pull/18279/commits/7a0f84fabbbd2e6cb1a5efdb867d224fc6de1992)